department-of-veterans-affairs/caseflow
# Backend changes for certification loading spinner

Connects https://github.com/department-of-veterans-affairs/caseflow/issues/2668

Makes backend changes that will allow us to show a spinner on Certification page load.

When v2 is enabled, the frontend will make a request to `/certifications/new/ID`, then just load a spinner and the React app. The react app should then poll `GET /certifications/ID`— it will return `{ data_loading: true }` until the job is finished, after which it will return the certification method

Testing:
- `bundle exec sidekiq`
- remove the bit about `perform_now` in dev mode
- insert `sleep 5` in the perform method StartCertificationJob
- go to `/certifications/new/123C` in a browser tab (it will fail)
- in another browser tab, verify that `/certifications/123C/json` returns `{ data_loading: true }` for a few seconds, then returns expected certification JSON after that

There is a followon issue here:
https://github.com/department-of-veterans-affairs/caseflow/issues/2707
